src/lib/common/include/sol-certificate.h

```diff
@@ -73,6 +84,36 @@ void sol_cert_unref(struct sol_cert *cert);
 const char *sol_cert_get_filename(const struct sol_cert *cert);
 
 /**
+ * @brief Read data from @a cert.
+ *
+ * @param cert Certificate object.
+ * @param buffer The buffer to write the read data.
+ *
+ * @return 0 on success or a negative error number on errors.
+ */
+int sol_cert_read_data(const struct sol_cert *cert, struct sol_buffer *buffer);
```

Comment:
no, we're working with blobs, see the api introduced by @ceolin. actually he offers this one already, just missing the "save/persist".